### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by madsmty**
Ultimos cambios solicitados

Factories
Quitar Unknowns
Pruebas con Mocha/Chai

### CODE REVIEW

File: src/controllers/logToDB.ts [hunk 1]
```diff
@@ -7,18 +7,38 @@ export class LogController {
...
+           
+
+            try {
+                decoded = jwt.verify(token, 'SECRET') as TokenInterface
+            }
+            catch{
+                decoded = {userRole:"Guest",userId:"NoUserId"}
+            }
+
+            /*
```
Comment (danielavaladezacc) [inline src/controllers/logToDB.ts]: Let's remove commented code and run prettier.

File: src/factories/posts.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { Posts } from '../models/posts'
+import { ServerPostArray } from '../interfaces/serverPostArray'
+import { ServerUserElement } from '../interfaces/serverUserElement'
+
+export const convertPosts = (
```
Comment (danielavaladezacc) [inline src/factories/posts.ts]: Let's create this as a factory file. For example:

```
export const postsFactory = {
    convertPosts(...) = {}
}
```

`postsFactory.convertPosts(...)`

File: src/factories/users.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { ServerUserArray } from '../interfaces/serverUserArray'
+import { User } from '../models/users'
+
+export const convertName = (paramArray: ServerUserArray): Array<User> => {
```
Comment (danielavaladezacc) [inline src/factories/users.ts]: Same thing here.

**Review by danielavaladezacc — CHANGES_REQUESTED** [review]
Let's apply unit tests to the factory files.

File: src/factories/posts.ts (hunk 2 above)
Comment (madsmty) [reply src/factories/posts.ts]: done

File: src/controllers/logToDB.ts (hunk 1 above)
Comment (madsmty) [reply src/controllers/logToDB.ts]: done

File: src/factories/users.ts (hunk 3 above)
Comment (madsmty) [reply src/factories/users.ts]: done

File: src/factories/posts.factory.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import { Posts } from '../models/posts'
+import { ServerPostArray } from '../interfaces/serverPostArray'
+import { ServerUserElement } from '../interfaces/serverUserElement'
+
+export class PostsFactory {
+     convertPosts = (
+        paramUser: ServerUserElement,
+        paramPostArray: ServerPostArray
+    ): Array<Posts> => {
```
Comment (danielavaladezacc) [inline src/factories/posts.factory.ts:9]: Change to `Posts[]`

File: src/factories/users.factory.ts [hunk 5]
```diff
@@ -1,36 +1,40 @@
...
-              prefix = nameSeparator[0]
-              firstName = nameSeparator[1]
-              lastName = nameSeparator[2]
-          } else {
-              prefix = ''
-              firstName = nameSeparator[0]
-              lastName = nameSeparator[1]
+export class UsersFactory
+    {
+        convertNames = (paramArray: ServerUserArray): Array<User> => {
```
Comment (danielavaladezacc) [inline src/factories/users.factory.ts]: Change to `User[]`

**Review by danielavaladezacc — APPROVED** [review]

**Pull request merged**